### PR TITLE
implemented merge sort for kotlin lang

### PR DESCRIPTION
**Pull request by gat786**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have.

### Description

Implementation of Merge Sort for Kotlin Lang

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

It is not enlisted in Issues. Atleast, not that I know of.

### CODE REVIEW

File: Kotlin/sort/MergeSort/src/MergeSort.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
+fun mergeSort(list : MutableList<Int>) : MutableList<Int>{
```
Comment (srimanikantapalakollu) [inline Kotlin/sort/MergeSort/src/MergeSort.kt:1]: @gat786 Add the Sample Output at the end of the program.

File: Kotlin/sort/MergeSort/src/MergeSort.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                sortedList[k] = leftSide[i]
+                i++
+            }
+            else {
+                sortedList[k] = rightSide[j]
+                j++
+            }
+            k++
+        }
+
```
Comment (srimanikantapalakollu) [inline Kotlin/sort/MergeSort/src/MergeSort.kt:33]: @gat786 Remove all the extra spaces in the code file.

**Review by srimanikantapalakollu — CHANGES_REQUESTED** [review]

File: Kotlin/sort/MergeSort/src/MergeSort.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    val list = MutableList(lengthOfList) { 0 }
+    list.withIndex().forEach {
+        print("Enter value ${it.index + 1} - ")
+        list[it.index] = readLine()!!.toInt()
+    }
+    println("unsorted list is $list")
+
+    val sorted = mergeSort(list.toMutableList());
+    println("sorted list is $sorted")
+}
```
Comment (vatsalkesarwani12) [inline Kotlin/sort/MergeSort/src/MergeSort.kt:71]: add space and time complexity

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
Are you working on the same @gat786 ?

**Comment by gat786** [comment]
Yes been a little busy the past week.

**Comment by gat786** [comment]
@HarshCasper I added the required changes and I hope the current version is acceptable

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**